### DISCUSSION TITLE
Streaming problem on ESP32 AUDIO KIT

### DISCUSSION THREAD

**Discussion by flashdato** [pschatzmann/arduino-audio-tools#1436] [Q&A] [answered]
I am using basic example code this library provides to stream music on my board. But when it connects to wifi it gives out this error and starts rebooting.

There is an error while running this stream link 

[I] AudioSource.h : 186 - selectStream: 0/0 -> https://dolphin-app-dh42m.ondigitalocean.app/audio/redhat.mp3
[I] URLStream.h : 86 - virtual bool audio_tools::URLStream::begin(const char*, const char*, MethodID, const char*, const char*): https://dolphin-app-dh42m.ondigitalocean.app/audio/redhat.mp3
[I] Url.h : 74 - Url::parse
[I] Url.h : 112 - url->https://dolphin-app-dh42m.ondigitalocean.app/audio/redhat.mp3
[I] Url.h : 113 - host->dolphin-app-dh42m.ondigitalocean.app
[I] Url.h : 114 - protocol->https
[I] Url.h : 115 - path->/audio/redhat.mp3
[I] Url.h : 116 - port->443
[I] URLStream.h : 417 - void audio_tools::URLStream::login()
[I] HttpRequest.h : 299 - Request written ... waiting for reply
[I] HttpHeader.h : 248 - Waiting for data...
[I] HttpHeader.h : 257 - Data available: 1369
CORRUPT HEAP: Bad head at 0x3ffbcbd8. Expected 0xabba1234 got 0x3ffb9a34

assert failed: multi_heap_free multi_heap_poisoning.c:259 (head != NULL)

Backtrace: 0x40083fa5:0x3ffb1c10 0x4008df95:0x3ffb1c30 0x40093b01:0x3ffb1c50 0x400936f9:0x3ffb1d80 0x40084515:0x3ffb1da0 0x40093b31:0x3ffb1dc0 0x4016c779:0x3ffb1de0 0x4016cd29:0x3ffb1e00 0x400dbc43:0x3ffb1e20 0x400dbc67:0x3ffb1e40 0x400dbc72:0x3ffb1e60 0x400dbf25:0x3ffb1e80 0x400dc081:0x3ffb1ec0 0x400dc1b3:0x3ffb1fa0 0x400dc314:0x3ffb2000 0x400dc60d:0x3ffb2040 0x400dc66f:0x3ffb2080 0x400d4eb7:0x3ffb20a0 0x400dc722:0x3ffb20d0 0x400dc880:0x3ffb2110 0x400dc970:0x3ffb2130 0x400d486d:0x3ffb2170 0x400ddb17:0x3ffb2190 0x400ddeff:0x3ffb21b0 0x400ec8f6:0x3ffb2290

ELF file SHA256: b235e91eff2fdd99

This is the streaming link u can test it works on computer normally:

https://dolphin-app-dh42m.ondigitalocean.app/audio/redhat.mp3

**Comment by pschatzmann** [comment]
Try to download the main version with the latest corrections and make sure that PSRAM is deactivated or roll back to the latest released version v0.9.7

**Reply by flashdato** [reply]
PSRAM is disabled.

I tried the newest version and v0.9.7 but the problems remain the same.

**Comment by flashdato** [comment]
Is it possible that my ESP32 is overloading? I had similar problems with other streaming links but in that case, it started and most of the time crushed on the 10-second mark.

**Reply by pschatzmann** [reply, marked as answer]
I could reproduce your issue and committed some more corrections.
I hope this will resolve the issue now

**Reply by flashdato** [reply]
Thank you so much it worked you are the best at this
